internal/handlers/common/aggregations/aggregations.go

```diff
@@ -22,8 +22,29 @@ import (
 	"github.com/FerretDB/FerretDB/internal/util/iterator"
 )
 
-// Stage is a common interface for all aggregation stages.
-type Stage interface {
+// ProcessorStage is a common interface for aggregation stages that process document iterator.
+type ProcessorStage interface {
 	// Process applies an aggregate stage on documents from iterator.
 	Process(ctx context.Context, iter types.DocumentsIterator, closer *iterator.MultiCloser) (types.DocumentsIterator, error)
 }
+
+// ProducerStage is a common interface aggregation stages produce documentiterator.
+type ProducerStage interface {
+	// Produce applies an aggregate stage on documents from iterator.
```

Comment:
That's not correct

Reply:
Indeed thanks, it should return a document iterator.